static/js/cf-frontend.js

```diff
@@ -346,15 +348,33 @@ function selectedCheckerOnChange() {
 
 
 }
+
+function linkGen(){
```

Comment:
Hi Ashley, before merging this PR, one tiny thing need to be polished:

Add a code doc of this new added function, format would like:

/** 
* One sentence of describing what this function does (no implementation details involved): e.g. generated permanent link of user code and selected type system.
* Given a example of what the link generated would looks like.
*/

Thanks!